server/src/main/java/org/elasticsearch/index/translog/TranslogConfig.java

```diff
@@ -143,4 +143,12 @@ public OperationListener getOperationListener() {
     public boolean fsync() {
         return fsync;
     }
+
+    /**
+     * @return true if the configuration allows the Translog files to exist.
+     */
+    public boolean hasTranslog() {
+        // Expect no translog files to exist for searchable snapshots
+        return false == indexSettings.getIndexMetadata().isSearchableSnapshot();
```

Comment:
I limited the scope to searchable snapshot shards (including closed searchable snapshot shards).